src/frontend/github-oauth.ts

```diff
@@ -148,14 +151,39 @@ export default (opts: OAuthOptions): GithubOAuth => {
 		checkGithubAuth: async (req: Request, res: Response, next: NextFunction) => {
 			const traceLogsEnabled = await booleanFlag(BooleanFlags.TRACE_LOGGING, false);
 			const tracer = new Tracer(logger.child(opts), "checkGithubAuth", traceLogsEnabled);
+			try {
+				const { githubToken } = req.session;
+				if (!githubToken) {
+					tracer.trace("github token missing, calling login()");
+					throw "Missing github token";
+				}
+				tracer.trace("found github token in session. validating token with API.");
 
-			if (!req.session.githubToken) {
-				tracer.trace("found github token in session, calling login()");
-				res.locals.redirect = req.originalUrl;
-				return login(req, res);
+				await axios.get(`https://api.${envVars.GITHUB_DOMAIN}`, {
```

Comment:
This is the call that will validate the token.  Returns 200 if good, 401 if not.